supervisor/process.py

```diff
@@ -168,7 +168,8 @@ def change_state(self, new_state, expected=True):
         self.state = new_state
         if new_state == ProcessStates.BACKOFF:
             now = time.time()
-            self.backoff += 1
+            if self.backoff < self.config.maxbackoff or self.config.maxbackoff == 0:
+                self.backoff += self.config.backoffincrement
```

Comment:
@cdelguercio to prevent the issue you mentioned in https://github.com/Supervisor/supervisor/pull/1626#issuecomment-1969883652 you can use `min`:
```
self.backoff = min(self.backoff + self.config.backoffincrement, self.config.maxbackoff)
```

Reply:
I think you'd have to do something like:
```
self.backoff = self.config.maxbackoff == 0 ? self.backoff + self.config.backoffincrement : min(self.backoff + self.config.backoffincrement, self.config.maxbackoff);
```
The question is: what's preferred?

Reply:
Yeah I see what you are saying, I missed the `self.config.maxbackoff == 0` case.

What you have suggested looks correct :+1: